lib/modules/manager/unity3d/extract.ts

```diff
@@ -0,0 +1,55 @@
+import { regEx } from '../../../util/regex';
+import type { PackageDependency, PackageFileContent } from '../types';
+
+export const fileMatchRegex: string[] = [
+  '(^|/)ProjectSettings/ProjectVersion.txt',
+];
+
+const supportedExtensionsKey = [
+  'm_EditorVersionWithRevision',
+  'm_EditorVersion',
+];
+const parseLine = (line: string): PackageDependency | null => {
+  const matches = regEx(/^(?<depName>.+): (?<currentValue>.+)/g).exec(line);
+  if (!matches) {
+    return null;
+  }
+  const key = matches.groups?.depName;
+  const version = matches.groups?.currentValue;
+  if (!key || !version || !supportedExtensionsKey.includes(key)) {
+    return null;
+  }
+
+  if (version !== version.trim()) {
+    return null;
+  }
+
+  return {
+    autoReplaceStringTemplate:
+      '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}',
+    currentDigest: undefined,
+    currentValue: version,
+    datasource: 'unity3d',
+    depName: key,
+    depType: 'final',
+    replaceString: matches[0],
```

Comment:
```suggestion
    currentValue: version,
    datasource: 'unity3d',
    depName: key,
    depType: 'final',
```
These vales are set by default. 

Also why `final` as `depType`, if there is only one?